### PR TITLE
libelement/c api improvements

### PR DESCRIPTION
**Pull request by GTOliver**

### CODE REVIEW

File: libelement/include/element/interpreter.h [hunk 1]
```diff
@@ -289,50 +308,50 @@ element_result element_interpreter_compile(
  * @return ELEMENT_ERROR_API_INVALID_INPUT inputs pointer is null
  * @return ELEMENT_ERROR_API_OUTPUT_IS_NULL outputs pointer is null
  */
-element_result element_interpreter_evaluate(
+element_result element_interpreter_evaluate_declaration(
```
Comment (Zephilinox) [inline libelement/include/element/interpreter.h:311]: this evaluates an instruction tree, not a declaration

File: libelement/include/element/object.h [hunk 2]
```diff
@@ -52,77 +54,77 @@ element_result element_delete_compilation_ctx(element_compilation_ctx** context)
...
+ * @return ELEMENT_ERROR_API_OBJECT_MODEL_CTX_IS_NULL context pointer is null
+ * @return ELEMENT_ERROR_API_OBJECT_IS_NULL unsimplified object pointer is null
  * @return ELEMENT_ERROR_API_OUTPUT_IS_NULL output is null
  */
-element_result element_object_compile(
-    const element_object* object,
-    element_compilation_ctx* context,
-    element_object** output);
+element_result element_object_simplify(
+        const element_object* object,
```
Comment (Zephilinox) [inline libelement/include/element/object.h:79]: what's the reason for double indentation on some of these changes? it makes it inconsistent with the current indentation

File: libelement/src/token_internal.hpp [hunk 3]
```diff
@@ -37,8 +37,8 @@ struct element_tokeniser_ctx
     element_token* get_token(std::size_t token_index, element_result& out_result);
 
     std::shared_ptr<element_log_ctx> logger = nullptr;
-    const char* raw_filename = nullptr;
-    std::string filename;
+    const char* raw_source_name = nullptr;
+    std::string source_name;
```
Comment (Zephilinox) [inline libelement/src/token_internal.hpp:41]: can you look at removing source_name entirely? I don't think it's something we want to use, the `raw_source_name` pointer is important to preserve right now.

File: libelement/include/element/object.h (hunk 2 above)
Comment (GTOliver) [reply libelement/include/element/object.h:79]: This wasn't a conscious choice - it's just how the IDE is configured by default and I must not have noticed. I can try and change my IDE settings to be consistent 👍 .

File: libelement/include/element/interpreter.h (hunk 1 above)
Comment (GTOliver) [reply libelement/include/element/interpreter.h:311]: 👍

**Review by Zephilinox — APPROVED** [review]

**Pull request merged**